f.el

```diff
@@ -483,9 +483,13 @@ detect the depth.
     byte-compile-current-file)
    (:else (buffer-file-name))))
 
+(defvar f--path-separator nil
+  "A variable to cache result of `f-path-separator'.")
+
 (defun f-path-separator ()
   "Return path separator."
-  (substring (f-join "x" "y") 1 2))
+  (or f--path-separator
+      (setf f--path-separator (substring (f-join "x" "y") 1 2))))
```

Comment:
Why not `setq`?

Reply:
I suppose there's no specific reason to prefer `setf` here. I just tend to use it everywhere to have a uniform concept of assignable place instead of remember whether I deal with variable or, say, first element of a pair. Should I amend that?

Reply:
Don't worry about it, just curious 🙂